### ISSUE TITLE
Support for @supports syntax

### ISSUE THREAD

**Issue by bmbrands** [sabberworm/PHP-CSS-Parser#127]
The @supports syntax used in Bootstrap 4 is not supported.

When trying to compile it an exception is thrown:

Stack trace:
line 113 of /lib/php-css-parser/Parser.php: Sabberworm\CSS\Parsing\SourceException thrown
line 197 of /lib/php-css-parser/Parser.php: call to Sabberworm\CSS\Parser->parseList()
line 119 of /lib/php-css-parser/Parser.php: call to Sabberworm\CSS\Parser->parseAtRule()
line 96 of /lib/php-css-parser/Parser.php: call to Sabberworm\CSS\Parser->parseListItem()
line 87 of /lib/php-css-parser/Parser.php: call to Sabberworm\CSS\Parser->parseList()
line 82 of /lib/php-css-parser/Parser.php: call to Sabberworm\CSS\Parser->parseDocument()
line 61 of /testcss.php: call to Sabberworm\CSS\Parser->parse()

Exception - Unexpected end of document [line no: 23]

**Comment by raxbg** [comment]
Seems to be working for me. Can you provide a more elaborated example?

**Comment by sabberworm** [comment]
@raxbg Maybe you’re parsing in lenient mode? `@supports` is currently unsupported so this issue is valid IMHO.

**Comment by raxbg** [comment]
Yes, I will probably work on this some time soon because I will need it 😄

**Comment by JakeQZ** [comment]
Yep, I find the issue is in strict mode only.  In lenient mode, `@media` and `@supports` are parsed correctly.  In strict mode, an exception is thrown for either (and also any at-rule with nested rules, such as `@keyframes` and `@document`).

E.g. with the current code base the following test added to `AtRuleBlockListTest` will fail:

```php
    /**
     * @test
     *
     * @param string $css
     *
     * @dataProvider mediaRuleDataProvider
     */
    public function parsesMediaRuleInStrictMode($css)
    {
        (new Parser($css, Settings::create()->beStrict()))->parse();

        self::assertTrue(true); // no exception was thrown
    }
```

The failure is:
```
Sabberworm\CSS\Parsing\UnexpectedTokenException: Identifier expected. Got “}” [line no: 1]
```

Perhaps it is simply that the final closing `}` isn't being `consume`d - it is clearly triggering the completion of the at-rule, otherwise subsequent rules would end up inside the at-rule (which they don't) - so the parser ends up seeing it again, but in lenient mode silently ignores this double-vision.

See also #75.

Note that `@font-face` is fine, but that doesn't have nested rules, just property-value pairs.
